### PR TITLE
Pulling row_conversion.cu over from cudf

### PR DESCRIPTION
**Pull request by hyperbolic2346**
This code should live here in this repository rather than cudf. Since there is a calm in the churn of that file, it seems like a good time to bring it over here. My plan:

1. update row_conversion.cu in jni this with this PR
2. update the plugin to use jni version of row_conversion.cu
3. remove row_conversion.cu in cudf.

### CODE REVIEW

**Review by jlowe — COMMENTED** [review]
Code that is here looks OK, but cudf's TableTest.fixedWidthRowsRoundTripWide should be ported over to the RowConversionTest in spark-rapids-jni, so when we remove the cudf versions (and the corresponding cudf tests) we don't lose test coverage.  I think there's only one other cudf row conversion test that is already covered in spark-rapids-jni's RowConversionTest.

File: src/main/java/com/nvidia/spark/rapids/jni/RowConversion.java [hunk 1]
```diff
@@ -121,6 +167,8 @@ public static Table convertFromRows(ColumnView vec, DType ... schema) {
   }
 
   private static native long[] convertToRows(long nativeHandle);
+  private static native long[] convertToRowsFixedWidthOptimized(long nativeHandle);
 
   private static native long[] convertFromRows(long nativeColumnView, int[] types, int[] scale);
+  private static native long[] convertFromRowsFixedWidthOptimized(long nativeColumnView, int[] types, int[] scale);
```
Comment (jlowe) [inline src/main/java/com/nvidia/spark/rapids/jni/RowConversion.java:173]: I see two new JNI methods being declared here but no corresponding changes to RowConversionJni.cpp.

Comment (hyperbolic2346) [reply src/main/java/com/nvidia/spark/rapids/jni/RowConversion.java:173]: Good catch. I believe I have this sorted, but I'm still having trouble testing myself.

**Comment by hyperbolic2346** [comment]
build

**Comment by jlowe** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**